server/src/api/api.ts

```diff
@@ -248,11 +305,131 @@ const resolvers: any = {
             ...args.updatedItem
         };
     },
-    categories: async (root, _args, _context): Promise<[Category]> => {
+    createRequest: async (root, _args, _context) => {
         // @ts-ignore
         const {args, context} = fixArguments(root, _args, _context);
-        return await DB.from("categories");
+        // if non-admin, user on request must be user signed in
+        // TODO: test this
+        if (!context.user.admin && context.user.uuid !== args.newRequest.user_id) {
+            throw new GraphQLError("Unable to create request because you are not an admin and your UUID " +
+                "does not match the UUID of the user this request is for");
+        }
+
+        // make sure the user the request is for exists
+        const users = await DB.from("users").where({
+            uuid: args.newRequest.user_id
+        }).select("name", "email", "uuid", "phone", "slackUsername", "haveID", "admin");
+
+        let user;
+        if (users.length === 0) {
+            throw new GraphQLError("Unable to create this request because no user with the UUID provided was found");
+        } else {
+            user = users[0];
+        }
+
+        // fetch the item
+        const item: any = await getItem(args.newRequest.request_item_id, context.user.admin);
```

Comment:
why is this any? does `Promise<?Item>` (or whatever the ts equiv is) work?

Reply:
tl;dr I need to fix #43